test/testcafe/spec/RequestActivationEmail_spec.js

```diff
@@ -0,0 +1,74 @@
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrRequestActivationEmail)
+  .onRequestTo('http://localhost:3000/idp/idx/request-activation')
+  .respond(xhrRequestActivationEmailSuccess);
+
+const activationEmailInvalidTokenMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrInvalidToken);
+
+const activationEmailExpiredTokenMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrExpiredToken);
+
+const activationEmailUserSuspendedMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrUserSuspended);
+
+fixture('Activation Email');
+
+async function setup(t) {
+  const requestActivaitonEmailPage = new RequestActivationEmailPageObject(t);
+  await requestActivaitonEmailPage.navigateToPage();
+  return requestActivaitonEmailPage;
+}
+
+async function setupTerminal(t) {
+  const terminalPageObject = new TerminalPageObject(t);
+  await terminalPageObject.navigateToPage();
+  await checkConsoleMessages({
+    controller: null,
+    formName: 'terminal',
+  });
+  return terminalPageObject;
+}
+
+test.requestHooks(requestActivationEmailMock)('should render error with action button', async t => {
+  const requestActivationEmailPage  = await setup(t);
+  await t.expect(requestActivationEmailPage.getFormTitle()).eql('Activation link has expired');
+  await t.expect(requestActivationEmailPage.getErrorBoxText()).eql('Your account activation link is no longer valid. Request a new activation email below.');
+  await t.expect(requestActivationEmailPage.getSaveButtonLabel()).eql('Request activation email');
+});
+
+test.requestHooks(activationEmailInvalidTokenMock)('should render error if activation is link invalid', async t => {
+  const terminalPage = await setupTerminal(t);
+  await t.expect(terminalPage.getFormTitle()).eql('Activation link no longer valid');
+  await t.expect(terminalPage.getErrorBoxText()).eql('This can happen if you have already activated your account, or if the URL you are trying to use is invalid. Contact your administrator for further assistance.');
+});
+
+test.requestHooks(activationEmailExpiredTokenMock)('should render error if activation link has expired', async t => {
+  const terminalPage = await setupTerminal(t);
+  await t.expect(terminalPage.getFormTitle()).eql('Activation link has expired');
+  await t.expect(terminalPage.getErrorBoxText()).eql('Your account activation link is no longer valid. Contact your administrator to request a new activation link.');
+});
+
+test.requestHooks(activationEmailUserSuspendedMock)('should render error if user is suspended', async t => {
+  const terminalPage = await setupTerminal(t);
+  await t.expect(terminalPage.getFormTitle()).eql('Account suspended');
+  await t.expect(terminalPage.getErrorBoxText()).eql('Your account has been temporarily suspended. Contact your administrator for further assistance.');
+});
```

Comment:
Should we add a test case for the happy path? i.e. testing the message: `New activation link requested. If your information matches our records, you will receive a new activation link in your inbox soon.`

Reply:
Good catch.. I forgot to add `click` on the button flow..